pkg/asset/installconfig/azure/session.go

```diff
@@ -98,11 +109,31 @@ func GetSessionWithCredentials(cloudName azure.CloudEnvironment, armEndpoint str
 		}
 	}
 	var cred azcore.TokenCredential
-	if credentials.ClientCertificatePath != "" {
-		logrus.Warnf("Using client certs to authenticate. Please be warned cluster does not support certs and only the installer does.")
+	var authType AuthenticationType
+	switch {
+	case credentials.ClientCertificatePath != "":
+		authKey := string(rune(ClientCertificateAuth))
+		if _, has := onceLoggers[authKey]; !has {
+			onceLoggers[authKey] = new(sync.Once)
+		}
+		onceLoggers[authKey].Do(func() {
+			logrus.Warnf("Using client certs to authenticate. Please be warned cluster does not support certs and only the installer does.")
+		})
 		cred, err = newTokenCredentialFromCertificates(credentials, cloudConfig)
-	} else {
+		authType = ClientCertificateAuth
+	case credentials.ClientSecret != "":
 		cred, err = newTokenCredentialFromCredentials(credentials, cloudConfig)
+		authType = ClientSecretAuth
+	default:
```

Comment:
The default behavior appears to have changed but the original card states no docs changes should be necessary. If this is true should we change the default or docs in this case?

Reply:
We might have to review the need for docs.